### PR TITLE
Setup of React build environment

### LINKED ISSUE #1: Setup of environment
In order to get started, the repo structure must be defined and setup. The following steps should be included/performed in this setup:

- [x] Create Node/React application structure with TypeScript
- [x] Tighten TypeScript settings
- [x] React StrictMode
- [x] Setup GitHub workflow (building, code style and linting, tests, npm deploy on release)
- [x] Testing framework: Jest + React testing library
- [x] eslint
- [x] Prettier
- [x] Webpack
- [x] Babel
- [x] Changelog file
- [x] Add scripts to `package.json` for publishing, testing and validating
- [x] Define folder and file structure
- [x] Change GitHub repo settings (do not allow to commit directly to `master`)

### PR DESCRIPTION
**Pull request by rubenthoms**
- Created React application structure with TypeScript
- Setup GitHub workflows (build, format, test, deploy and verify changelog changes)
- Installed testing framework (Jest + React-testing-library)
- eslint + prettier
- Setup and configured Webpack 5 (using Babel)
- Created changelog file
- Added scripts for publishing, testing, formatting, validating and transpiling

Closes #1.

### CODE REVIEW

File: .github/workflows/flow-diagram-explorer.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: 🕵️ Check code style & linting
+        run: |
+          npm run typecheck && npm run lint
+
+      - name: 🤖 Run tests
+        run: |
+          npm run test
+
+      - name: 🔼 Build and publish Node.js package
+        if: github.event_name == 'release' && matrix.python-version == '3.6'
```
Comment (anders-kiaer) [inline .github/workflows/flow-diagram-explorer.yml:41]:
```suggestion
        if: github.event_name == 'release'
```

File: .github/workflows/check-changelog.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: check changelog
```
Comment (jsolaas) [inline .github/workflows/check-changelog.yml:1]: Nice

**Review by jsolaas — APPROVED** [review]
Looks good!

File: package.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
+{
+  "name": "flow-diagram-explorer",
```
Comment (anders-kiaer) [inline package.json:2]: @rubenthoms @jsolaas You can decide this later of course, but could perhaps be natural to put the npm package under the Equinor scope? https://www.npmjs.com/org/equinor

**Pull request merged**